controller/internal/routes/config_api_model.go

```diff
@@ -125,27 +125,47 @@ func MapConfigToRestModel(ae *env.AppEnv, config *model.Config) (*rest_model.Con
 	return ret, nil
 }
 
-func narrowJsonTypes(m map[string]interface{}) {
-	for k, v := range m {
-		if parsedNumber, ok := v.(ParsedNumber); ok {
-			//floats don't parse as int, try int fist, then float, else give up
-			if intVal, err := parsedNumber.Int64(); err == nil {
-				v = intVal
-				m[k] = intVal
-			} else if floatVal, err := parsedNumber.Float64(); err == nil {
-				v = floatVal
-				m[k] = floatVal
+func narrowJsonType(v *interface{}) {
+	if parsedNumber, ok := (*v).(ParsedNumber); ok {
+		//floats don't parse as int, try int first, then float, else give up
+		if intVal, err := parsedNumber.Int64(); err == nil {
+			*v = intVal
+		} else if floatVal, err := parsedNumber.Float64(); err == nil {
+			*v = floatVal
+			intVal := math.Trunc(floatVal)
+			if intVal == floatVal {
+				*v = intVal
 			}
 		}
+	}
+}
+
+func narrowJsonTypesList(l []interface{}) {
+	for i, v := range l {
+		narrowJsonType(&v)
 
 		switch val := v.(type) {
-		case float64:
```

Comment:
I don't really understand why this was being done here instead of when parsedNumber.Float64() succeeded. @andrewpmartinez

Reply:
Because the above section is gated by a type assertion check and it may or may not be hit.

IIRC this was to deal with how `go-swagger` was dealing with maps that contained values where it explicitly didn't know the type to give to a number because the config type definition is defined as "a map of things to values" which denotes no specific types for number-like values.

`go-swagger` instead could either guess or give us an interface object w/ methods to figure it out on our own. The below interface `ParsedNumber` is a sub-interface of that interface and used to detect that scenario.

Reply:
Per my comment above I don't know if anything changed that makes it so we don't need to look for `ParsedNumber` anymore. 

This new logic assumes we have `int64`/`float64` every time and not `ParsedNumber`.